### PR TITLE
HOTFIX: Apply key-guessing heuristic to more places

### PR DESCRIPTION
**Pull request by danielballan**
Following up on #281, a little less hack-ish. Bluesky looks at the *order* of `read_attrs` and assumes the first one is the one it should use to (1) compute relative trajectories and (2) plot against in the subscription factories in `simple_scans.py`.

### CODE REVIEW

File: bluesky/utils.py [hunk 1]
```diff
@@ -387,3 +387,29 @@ def snake_cyclers(cyclers, snake_booleans):
             expanded = v2[:total_length]
             new_cyclers.append(cycler(k, expanded))
     return reduce(operator.add, new_cyclers)
+
+
+def first_key_heuristic(device):
+    """
+    Get the fully-qualified data key for the first entry in read_attrs.
+
+    This will raise is that entry is not a Signal.
```
Comment (tacaswell) [inline bluesky/utils.py]: will raise if entry has not exactly 1 key in describe.

Comment (danielballan) [inline bluesky/utils.py]: True, I shouldn't refer to `Signal` here.

**Comment by klauer** [comment]
I think this is the path of least resistance, so for now it's good enough.

A minor side-note: `Device.read` in ophyd returns `OrderedDict`s whereas `Signal.read` returns a dict. Maybe we should be more strict about that API - then `read_attrs` would not need to be checked by bluesky.

**Comment by danielballan** [comment]
@klauer That sounds good. Please merge?

**Pull request merged**